scripts/release.py

```diff
@@ -402,6 +458,38 @@ def request_doc_update(github_token, repo, release):
 
 def build_parser():
     parser = argparse.ArgumentParser(description='Publish RPCO tag.')
+    # Mandatory args
+    parser.add_argument(
+        '--tag', type=Tag, required=True, help='Name of new tag.'
+    )
+    existing = parser.add_mutually_exclusive_group(required=True)
+    existing.add_argument(
+        '--commit', help='Reference to the commit to tag.'
+    )
+    existing.add_argument(
+        '--existing-release', default=False, action='store_true'
+    )
+    # Recommended args
+    parser.add_argument(
+        '--future-tag', type=Tag,
+        help='The version number for the next cycle'
+    )
+    # Convenience args that defaults to ENV vars
+    parser.add_argument(
+        '--branch', help='Branch to update'
```

Comment:
For point 1: The creation of the tag can happen in any branch and is not branch related. So we're fine there.
For point 2: Yes. That's the goal. 

If a commit "b" happens between the commit tagged ("a") and the new commit pushed ("c"), we can say the commit "b" was built using the old release version and slipped through the cracks.

The risk is very low, because:
1. The later commits ("d" and so on..) will be tested, and if "b" broke the build, we'll see it immediately.
2. Whether or not it breaks the build, the commit "b" can be applied with the artifacts of "a". Else it would not be able to merge during this race condition.
3. Tagging/releasing for the past shouldn't use this process, or at least should be done with the proper flags. I assume here a minimum knowledge of the release process and what has to be done. This script is only used by @alextricity25 for now, and maybe in the distant future, to a CI bot.

We can still refine later.